lib/common/socket.c

```diff
@@ -277,6 +277,7 @@ const char *decode_ssl_input(h2o_socket_t *sock)
         { /* call SSL_read (while detecting SSL renegotiation and reporting it as error) */
             int did_write_in_read = 0;
             sock->ssl->did_write_in_read = &did_write_in_read;
+            ERR_clear_error();
```

Comment:
Perhaps this can be moved outside the loop to be called just once (although the overhead of `ERR_clear_error()` should be minimal.